akka-stream/src/main/scala/akka/stream/io/Framing.scala

```diff
@@ -0,0 +1,173 @@
+package akka.stream.io
+
+import java.nio.ByteOrder
+
+import akka.stream.scaladsl.Flow
+import akka.stream.stage.{ TerminationDirective, PushPullStage, SyncDirective, Context }
+import akka.util.{ ByteIterator, ByteStringBuilder, ByteString }
+
+import scala.annotation.tailrec
+
+object Framing {
+
+  def delimiter(delimiterBytes: ByteString, maximumFrameLength: Int): Flow[ByteString, ByteString, Unit] =
+    Flow[ByteString].transform(() ⇒ new DelimiterFramingStage(delimiterBytes, maximumFrameLength))
+      .named("delimiterFraming")
+
+  def lines(delimiter: String, maximumLineBytes: Int): Flow[ByteString, String, Unit] =
+    Framing.delimiter(ByteString(delimiter), maximumLineBytes).map(_.utf8String)
```

Comment:
Yes, I agree, assuming UTF8 encoding and not splitting on half codepoints may be good enough for now.